engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java

```diff
@@ -835,31 +834,37 @@ private Table headOrTailBy(long nRows, boolean head, String... groupByColumns) {
             String colName = colNames.get(i);
             if (!groupByColsSet.contains(colName)) {
                 final Class<?> dataType = getDefinition().getColumn(colName).getDataType();
-                casting[j] = colName + " = " + getCastFormula(dataType) + colName;
+                casting[j] = String.format("%s = %s%s", colName, getCastFormula(dataType), colName);
```

Comment:
Formats are nice, but they are definitely much more expensive than concatenation. This loop isn't so tight, though, so I suppose it's up to you if you want to keep this change or revert it.

Reply:
How do we usually decide between the two: performance vs readability?
Is that why we always keep string.format in exceptions because performance is not a concern there?